### PR TITLE
Fix: handle empty body in model_dump function

### PR DESCRIPTION
**Pull request by KonstantinVasilkov**
Fix issue in cadwyn/_compat.py in model_dump: previously if we pass empty body to the endpoint while trying to dump it, Cadwyn was throwing an error.

### CODE REVIEW

**Comment by zmievsa** [comment]
@KonstantinVasilkov instead of using "isinstnace dict", let's use "not isinstance(obj, BaseModel)".

It would handle a lot more cases.

File: cadwyn/_compat.py [hunk 1]
```diff
@@ -118,8 +118,8 @@ def model_fields(model: type[BaseModel]) -> dict[str, FieldInfo]:
         return model.__fields__  # pyright: ignore[reportDeprecated]
 
 
-def model_dump(model: BaseModel, by_alias: bool = False, exclude_unset: bool = False) -> dict[str, Any]:
-    if isinstance(model, dict):
+def model_dump(model: BaseModel, by_alias: bool = False, exclude_unset: bool = False) -> dict[str, Any] | BaseModel:
```
Comment (zmievsa) [inline cadwyn/_compat.py:121]:
```suggestion
def model_dump(model: Any | BaseModel, by_alias: bool = False, exclude_unset: bool = False) -> Any:
```

File: cadwyn/_compat.py [hunk 2]
```diff
@@ -118,8 +118,8 @@ def model_fields(model: type[BaseModel]) -> dict[str, FieldInfo]:
...
+    if not isinstance(model, BaseModel):
```
Comment (zmievsa) [inline cadwyn/_compat.py:122]:
```suggestion
    # it means we have a dict or a list instead of a full model
    if not isinstance(model, BaseModel):
```

**Comment by zmievsa** [comment]
Please, add a test that covers your scenario. It must be failing without your fix

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,10 @@ Please follow [the Keep a Changelog standard](https://keepachangelog.com/en/1.0.
 
 ## [Unreleased]
 
+## [3.4.1]
+### Fixed
+* model_dump now correctly handles empty payload and returns an empty dict instead of raising an exception
```
Comment (zmievsa) [inline CHANGELOG.md:10]: model_dump is a private interface. Describe which **use case** was fixed

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,10 @@ Please follow [the Keep a Changelog standard](https://keepachangelog.com/en/1.0.
 
 ## [Unreleased]
 
+## [3.4.1]
```
Comment (zmievsa) [inline CHANGELOG.md:8]: Please, run markdownlint on this file

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,12 @@ Please follow [the Keep a Changelog standard](https://keepachangelog.com/en/1.0.
...
+
+### Fixed
+
+* The empty body in POST request caused an exception on a try to serialize it. Now it is handled correctly
```
Comment (zmievsa) [inline CHANGELOG.md:12]:
```suggestion
* If the endpoint specified a single non-pydantic (list/dict) body parameter, Cadwyn failed to serialize the body
```

File: cadwyn/structure/versions.py [hunk 6]
```diff
@@ -529,6 +529,11 @@ async def _convert_endpoint_kwargs_to_version(
             raw_body: BaseModel | None = kwargs.get(body_field_alias)
             if raw_body is None:
                 body = None
+            # It means we have a dict or a list instead of a full model.
+            # This is a case when we use POST request to perform action rather than to create a new object,
+            # and we need to use "payload: dict = Body(None)" in a route.
```
Comment (zmievsa) [inline cadwyn/structure/versions.py:534]:
```suggestion
            # This covers the following use case in the endpoint definition: "payload: dict = Body(None)"
```

File: tests/test_data_migrations.py [hunk 7]
```diff
@@ -461,6 +461,22 @@ def migrator(request: RequestInfo):
             "bar": None,
         }
 
+    def test__model_dump_correctly_works_with_empty_request_body__should_return_empty_dict(
```
Comment (zmievsa) [inline tests/test_data_migrations.py:464]:
```suggestion
    def test__serialization_of_request_body__when_body_is_non_pydantic(
```

**Review by zmievsa — APPROVED** [review]
Added a few final minor suggestions. Big thanks for finding the bug and fixing it!

File: CHANGELOG.md (hunk 3 above)
Comment (KonstantinVasilkov) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 4 above)
Comment (KonstantinVasilkov) [reply CHANGELOG.md:8]: Fixed

**Pull request merged**